src/invert/laplacexy/laplacexy.cxx

```diff
@@ -214,6 +428,67 @@ LaplaceXY::LaplaceXY(Mesh *m, Options *opt, const CELL_LOC loc)
   // Now communicate to fill guard cells
   localmesh->communicate(indexXY);
 
+  // Fill corner guard cells by getting the y-guard cells from the x-neighbours.
+  // This is an arbitrary choice, which makes a difference at the X-point, where taking
+  // the x-guard cells from the y-neighbours would give points in different regions.
+  auto xcomm = localmesh->getXcomm();
+  int proc_xind = localmesh->getXProcIndex();
+  int tag0 = localmesh->getYProcIndex()*localmesh->getNXPE();
+  MPI_Request requests[] = {MPI_REQUEST_NULL, MPI_REQUEST_NULL};
+  MPI_Status statuses[2];
+
+  // Get lower, inner corner
+  if (not localmesh->firstX()) {
+    // Receive from inner
+    MPI_Irecv(&indexXY(localmesh->xstart-1, localmesh->ystart-1), 1, MPI_DOUBLE,
+        proc_xind-1, tag0 + 1*proc_xind, xcomm, &requests[0]);
+  }
+  if (not localmesh->lastX()) {
+    // Need to send lower outer point out
+    MPI_Isend(&indexXY(localmesh->xend, localmesh->ystart-1), 1, MPI_DOUBLE, proc_xind+1,
+        tag0 + 1*(proc_xind + 1), xcomm, &requests[1]);
+  }
+  MPI_Waitall(2, requests, statuses);
+
+  // Get upper, inner corner
+  if (not localmesh->firstX()) {
+    // Receive from inner
+    MPI_Irecv(&indexXY(localmesh->xstart-1, localmesh->yend+1), 1, MPI_DOUBLE,
+        proc_xind-1, tag0 + 1*proc_xind, xcomm, &requests[0]);
+  }
+  if (not localmesh->lastX()) {
+    // Need to send upper outer point out
+    MPI_Isend(&indexXY(localmesh->xend, localmesh->yend+1), 1, MPI_DOUBLE, proc_xind+1,
+        tag0 + 1*(proc_xind + 1), xcomm, &requests[1]);
+  }
+  MPI_Waitall(2, requests, statuses);
+
+  // Get lower, outer corner
+  if (not localmesh->lastX()) {
+    // Receive from outer
+    MPI_Irecv(&indexXY(localmesh->xend+1, localmesh->ystart-1), 1, MPI_DOUBLE,
+        proc_xind+1, tag0 + proc_xind, xcomm, &requests[0]);
+  }
+  if (not localmesh->firstX()) {
+    // Need to send lower inner point in
+    MPI_Isend(&indexXY(localmesh->xstart, localmesh->ystart-1), 1, MPI_DOUBLE,
+        proc_xind-1, tag0 + (proc_xind - 1), xcomm, &requests[1]);
+  }
+  MPI_Waitall(2, requests, statuses);
+
+  // Get upper, outer corner
+  if (not localmesh->lastX()) {
+    // Receive from outer
+    MPI_Irecv(&indexXY(localmesh->xend+1, localmesh->yend+1), 1, MPI_DOUBLE,
+        proc_xind+1, tag0 + proc_xind, xcomm, &requests[0]);
+  }
+  if (not localmesh->firstX()) {
+    // Need to send upper inner point in
+    MPI_Isend(&indexXY(localmesh->xstart, localmesh->yend+1), 1, MPI_DOUBLE,
+        proc_xind-1, tag0 + (proc_xind - 1), xcomm, &requests[1]);
+  }
+  MPI_Waitall(2, requests, statuses);
```

Comment:
Can the above set of 4 waits be merged, so have 8 requests and one `MPI_Waitall`?